src/app/icd/ICDManager.cpp

```diff
@@ -98,11 +104,60 @@ bool ICDManager::SupportsFeature(Feature feature)
     bool success        = (Attributes::FeatureMap::Get(kRootEndpointId, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
     return success ? ((featureMap & to_underlying(feature)) != 0) : false;
 #else
-
     return ((mFeatureMap & to_underlying(feature)) != 0);
 #endif // !CONFIG_BUILD_FOR_HOST_UNIT_TEST
 }
 
+void ICDManager::SendCheckInMsgs()
+{
+#if !CONFIG_BUILD_FOR_HOST_UNIT_TEST
+    VerifyOrDie(mStorage != nullptr);
+    VerifyOrDie(mFabricTable != nullptr);
+    for (const auto & fabricInfo : *mFabricTable)
+    {
+        uint16_t supported_clients = ICDManagementServer::GetInstance().GetClientsSupportedPerFabric();
+
+        ICDMonitoringTable table(*mStorage, fabricInfo.GetFabricIndex(), supported_clients /*Table entry limit*/,
+                                 mSymmetricKeystore);
+        if (!table.IsEmpty())
+        {
+            Messaging::ExchangeManager * exchangeManager = InteractionModelEngine::GetInstance()->GetExchangeManager();
+
+            for (uint16_t i = 0; i < table.Limit(); i++)
+            {
+                ICDMonitoringEntry entry(mSymmetricKeystore);
+                if (CHIP_NO_ERROR != table.Get(i, entry))
+                {
+                    // No more entries in the table
+                    break;
+                }
+
+                bool active =
+                    InteractionModelEngine::GetInstance()->IsMatchingSubscriptionActive(entry.fabricIndex, entry.monitoredSubject);
+
+                if (active)
+                {
+                    continue;
+                }
+
+                ICDCheckInSender * sender = mICDSenderPool.CreateObject(exchangeManager);
+
+                if (sender == nullptr)
+                {
+                    ChipLogError(AppServer, "Failed to allocate ICDCheckinSender");
+                    return;
+                }
+
+                if (CHIP_NO_ERROR != sender->RequestResolve(entry, mFabricTable))
+                {
+                    ChipLogError(AppServer, "Failed to send ICD Check-In");
+                }
```

Comment:
can be simplified with a VerifyOrReturn.

Reply:
Not here since we do want to iterate through all the ICD Monitoring entry and no simply exit the function directly.